src/sentry/migrations/0732_backfill_group_info_to_group_attributes.py

```diff
@@ -0,0 +1,206 @@
+# Generated by Django 5.0.6 on 2024-06-18 18:39
+
+from __future__ import annotations
+
+import dataclasses
+from datetime import datetime
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from django.conf import settings
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import F, Window
+from django.db.models.functions import Rank
+from sentry_kafka_schemas.schema_types.group_attributes_v1 import GroupAttributesSnapshot
+
+from sentry.issues.attributes import produce_snapshot_to_kafka
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils import redis
+from sentry.utils.iterators import chunked
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBarApprox
+
+if TYPE_CHECKING:
+    from sentry.models.group import Group
+    from sentry.models.groupassignee import GroupAssignee
+    from sentry.models.groupowner import GroupOwner
+
+CHUNK_SIZE = 10000
+
+
+class GroupOwnerType(Enum):
+    SUSPECT_COMMIT = 0
+    OWNERSHIP_RULE = 1
+    CODEOWNERS = 2
+
+
+@dataclasses.dataclass
+class GroupValues:
+    id: int
+    project_id: int
+    status: int
+    substatus: int | None
+    first_seen: datetime
+    num_comments: int
+    priority: int | None
+    first_release_id: str | None
+
+
+def _bulk_retrieve_group_values(group_ids: list[int], Group: type[Group]) -> list[GroupValues]:
+    group_values_map = {
+        group["id"]: group
+        for group in Group.objects.filter(id__in=group_ids).values(
+            "id",
+            "project_id",
+            "status",
+            "substatus",
+            "first_seen",
+            "num_comments",
+            "priority",
+            "first_release",
+        )
+    }
+    assert len(group_values_map) == len(group_ids)
+
+    results = []
+    for group_id in group_ids:
+        group_values = group_values_map[group_id]
+        results.append(
+            GroupValues(
+                id=group_id,
+                project_id=group_values["project_id"],
+                status=group_values["status"],
+                substatus=group_values["substatus"],
+                first_seen=group_values["first_seen"],
+                num_comments=group_values["num_comments"],
+                priority=group_values["priority"],
+                first_release_id=(
+                    str(group_values["first_release"]) if group_values["first_release"] else None
+                ),
```

Comment:
Why do we need to cast to a string here? Don't we just want the id as an int?

Reply:
The column type was wrong earlier - this should work as an int now